libs/pageserver_api/src/models.rs

```diff
@@ -227,6 +231,56 @@ pub struct TimelineInfo {
     pub state: TimelineState,
 }
 
+#[derive(Debug, Clone, Serialize)]
+pub struct LayerMapInfo {
+    pub in_memory_layers: Vec<InMemoryLayerInfo>,
+    pub historic_layers: Vec<HistoricLayerInfo>,
+}
+
+#[serde_as]
+#[derive(Debug, Clone, Serialize)]
+#[serde(tag = "kind")]
+pub enum InMemoryLayerInfo {
+    Open {
+        #[serde_as(as = "DisplayFromStr")]
+        lsn_start: Lsn,
+    },
+    Frozen {
+        #[serde_as(as = "DisplayFromStr")]
+        lsn_start: Lsn,
+        #[serde_as(as = "DisplayFromStr")]
+        lsn_end: Lsn,
+    },
+}
+
+#[serde_as]
+#[derive(Debug, Clone, Serialize)]
+#[serde(tag = "kind")]
+pub enum HistoricLayerInfo {
+    Delta {
+        layer_file_name: String,
+        #[serde_as(as = "DisplayFromStr")]
+        key_start: Key,
```

Comment:
Unsure if key needs to be here, we do have the `layer_file_name` already.

Reply:
Yeah, unless there's s a particular need to have the keys repeated, I don't see why we need them in `LayerMapInfo`.
If we don't need them, we can move back the `Key` struct into the `pageserver` crate.

Reply:
We can go with `LayerFileName` alone and remove the rest of the fields.
Yet, it has to be changed, since has `Range` in its fields.

Reply:
Moved `LayerFileName`, had to leave `Key` for that too, but now the API is better.